File: server/csharp/MetaObjects.Render/Extract/FieldExtraction.cs
namespace MetaObjects.Render.Extract;

/// <summary>
/// FROZEN cross-port per-field extraction classification. Names match the conformance corpus
/// expected.json exactly — the runner uses <c>.ToString()</c> comparison.
/// Do not reorder or rename without an ADR.
/// </summary>
public enum FieldExtraction
{
    EXTRACTED,
    // DEFAULTED is reserved (a future @default-backed value); the current engine does not emit it.
    DEFAULTED,
    LOST_OPTIONAL,
    LOST_REQUIRED,
    MALFORMED,
}

File: server/java/render/src/main/java/com/metaobjects/render/extract/FieldExtraction.java
package com.metaobjects.render.extract;

/** FROZEN cross-port per-field extraction classification. Do not reorder or add without an ADR. */
public enum FieldExtraction {
    EXTRACTED,
    // FR-011: a value reached via @coerceDefault (present-but-uncoercible fallback) or
    // @default (absent-fill). Classified by Extract via the terminal coercion kind.
    DEFAULTED,
    LOST_OPTIONAL,
    LOST_REQUIRED,
    MALFORMED
}
